Port most of the OAUTH2 support from Trino to Presto (#24443)
prestodb/presto: presto-docs/src/main/sphinx/security/oauth2.rst

========================
Oauth 2.0 Authentication
========================

> Presto can be configured to enable frontend OAuth2 authentication over HTTPS for clients such as the CLI, JDBC, and ODBC drivers. OAuth2 provides a secure and flexible way to authenticate users via an external identity provider (IdP), such as Okta, Auth0, Azure AD, or Google.

Reviewer: ```suggestion
Presto can be configured to enable frontend OAuth2 authentication over HTTPS for clients such as the CLI, JDBC, and ODBC drivers. OAuth2 provides a secure and flexible way to authenticate users by using an external identity provider (IdP), such as Okta, Auth0, Azure AD, or Google.
```
Avoid Latin terms and abbreviations, because they don't translate well and can be challenging for an international audience. See the [GitLab documentation style guide recommended word list entry for via](https://docs.gitlab.com/development/documentation/styleguide/word_list/#via) for alternatives.

> OAuth2 authentication in Presto uses the Authorization Code Flow with PKCE and OpenID Connect (OIDC). The Presto coordinator initiates an OAuth2 challenge, and the client completes the flow by obtaining an access token from the IdP.

Reviewer: I know that you just (properly) introduced IdP in the previous paragraph, but consider writing out "identity provider" here for ease of reading.

> Presto Server Config

Reviewer: ```suggestion
Presto Server Configuration
```

> --------------------

Reviewer: ```suggestion
---------------------------
```

> To enable OAuth2 authentication, configuration changes are made **only on the Presto coordinator**. No changes are required on the workers.

Reviewer: This is a great thing to warn the reader about, and it's in a highly visible location to reduce the chance of the reader overlooking it. Nice!

Secure Communication
--------------------

> Access to the Presto coordinator must be secured via HTTPS. You must configure a valid TLS certificate and keystore on the coordinator. See the `TLS setup guide <https://trino.io/docs/current/security/tls.html>`_ for details.

Reviewer: ```suggestion
Access to the Presto coordinator must be secured with HTTPS. You must configure a valid TLS certificate and keystore on the coordinator. See the `TLS setup guide <https://trino.io/docs/current/security/tls.html>`_ for details.
```
With, by using, through. See the [GitLab documentation style guide recommended word list entry for via](https://docs.gitlab.com/development/documentation/styleguide/word_list/#via).

> =============

Reviewer: Please edit lines 1 and 3 to be the same number of characters as line 2.

> =============

Reviewer: In order for this new page to show up in the Presto doc, it must be added to the table of contents page for the topic. 

Please add 
`security/oauth2`
to 
https://github.com/prestodb/presto/blob/master/presto-docs/src/main/sphinx/security.rst

> =============

Reviewer: This page seems unfinished. Re-request a review from me when there's content and I'm happy to review it then.

Oauth 2.0 Authentication
=============

Presto server configuration
---------------------------

Refresh Tokens

> ---------------------------

Reviewer: Please edit this line to be the same number of characters as line 8.
